src/permission/config-permission-handler.js

```diff
@@ -0,0 +1,243 @@
+var configValidator = require( './config-validator' );
+var configCompiler = require( './config-compiler' );
+var rulesMap = require( './rules-map' );
+var RuleApplication = require( './rule-application' );
+var RuleCache = require( './rule-cache' );
+var events = require( 'events' );
+var utils = require( 'util' );
+var fs = require( 'fs' );
+var STRING = 'string';
+
+/**
+ * A permission handler that reads a rules config JSON, validates
+ * its contents, compiles it and executes the permissions that it contains
+ * against every incoming message.
+ *
+ * This is the standard permission handler that deepstream exposes, in conjunction
+ * with the default permission.json it allows everything, but at the same time provides
+ * a convenient starting point for permission declarations.
+ *
+ * @author deepstreamHub GmbH
+ * @license [https://github.com/deepstreamIO/deepstream.io/blob/master/LICENSE] MIT
+ *
+ * @constructor
+ * @extends {EventEmitter}
+ *
+ * @param {Object} options deepstream options
+ * @param {[Object]} config  Optional config. If no config is provided, the ConfigPermissionHandler will attempt
+ *                           to load it from the path provided in options.permissionConfigPath.
+ */
+var ConfigPermissionHandler = function( options, config ) {
+	this._ruleCache = new RuleCache( options );
+	this._options = options;
+	this._config = null;
+	this.isReady = false;
+
+	if( config ) {
+		this.useConfig( config );
+	} else {
+		this.loadConfig( this._options.permissionConfigPath );
+	}
+};
+
+utils.inherits( ConfigPermissionHandler, events.EventEmitter );
+
+/**
+ * Temporary, will only be here until #146 (https://github.com/deepstreamIO/deepstream.io/issues/146)
+ * is resolved
+ *
+ * @param   {[type]}   connectionData [description]
+ * @param   {[type]}   authData       [description]
+ * @param   {Function} callback       [description]
+ *
+ * @returns {Boolean}
+ */
+/* istanbul ignore next */
+ConfigPermissionHandler.prototype.isValidUser = function( connectionData, authData, callback ) {
+	callback( null, authData.username || 'open' );
+};
+
+/**
+ * Load a configuration file. This will either load a configuration file for the first time at
+ * startup or reload the configuration at runtime
+ *
+ * @todo expose this method via the command line interface
+ *
+ * @param   {String} path the filepath of the permission.json file
+ *
+ * @public
+ * @cli loadConfig
+ * @returns {void}
+ */
+ConfigPermissionHandler.prototype.loadConfig = function( path ) {
+	fs.readFile( path, 'utf8', this._onConfigLoaded.bind( this, path ) );
+};
+
+/**
+ * Validates and compiles a loaded config. This can be called as the result
+ * of a config being passed to the permissionHandler upon initialisation,
+ * as a result of loadConfig or at runtime
+ *
+ * @todo expose this method via the command line interface
+ *
+ * @param   {Object} config deepstream permissionConfig
+ *
+ * @public
+ * @cli useConfig
```

Comment:
fair enough, I'll just make it a comment to keep track of what we want to expose via the command line interface